src/query_plan/fetch_dependency_graph_processor.rs

```diff
@@ -1,2 +1,395 @@
+            variable_definitions,
+            fragments,
+            operation_name,
+            assigned_defer_labels,
+            counter: 0,
+        }
+    }
+}
+
+impl FetchGroupProcessor<Option<PlanNode>, DeferredDeferBlock>
+    for FetchDependencyGraphToQueryPlanProcessor
+{
+    fn on_fetch_group(
+        &mut self,
+        group: &mut FetchDependencyGraphNode,
+        handled_conditions: &Conditions,
+    ) -> Result<Option<PlanNode>, FederationError> {
+        let op_name = self.operation_name.as_ref().map(|name| {
+            let counter = self.counter;
+            self.counter += 1;
+            let subgraph = to_valid_graphql_name(&group.subgraph_name);
+            format!("{name}__{subgraph}__{counter}")
+        });
+        Ok(group.to_plan_node(
+            &self.config,
+            handled_conditions,
+            &self.variable_definitions,
+            self.fragments.as_ref(),
+            op_name,
+        ))
+    }
+
+    fn on_conditions(
+        &mut self,
+        conditions: &Conditions,
+        value: Option<PlanNode>,
+    ) -> Option<PlanNode> {
+        let mut value = value?;
+        match conditions {
+            Conditions::Boolean(condition) => {
+                // Note that currently `ConditionNode` only works for variables
+                // (`ConditionNode.condition` is expected to be a variable name and nothing else).
+                // We could change that, but really, why have a trivial `ConditionNode`
+                // when we can optimise things righ away.
+                condition.then_some(value)
+            }
+            Conditions::Variables(variables) => {
+                for (name, negated) in &variables.0 {
+                    let (if_clause, else_clause) = if *negated {
+                        (None, Some(value))
+                    } else {
+                        (Some(value), None)
+                    };
+                    value = PlanNode::Condition(Arc::new(ConditionNode {
+                        condition_variable: name.clone(),
+                        if_clause,
+                        else_clause,
+                    }));
+                }
+                Some(value)
+            }
+        }
+    }
+
+    fn reduce_parallel(&mut self, values: &[Option<PlanNode>]) -> Option<PlanNode> {
+        flat_wrap_nodes(NodeKind::Parallel, values)
+    }
+
+    fn reduce_sequence(&mut self, values: &[Option<PlanNode>]) -> Option<PlanNode> {
+        flat_wrap_nodes(NodeKind::Sequence, values)
+    }
+
+    fn reduce_deferred(
+        &mut self,
+        defer_info: &DeferredInfo,
+        node: Option<PlanNode>,
+    ) -> Result<DeferredDeferBlock, FederationError> {
+        Ok(DeferredDeferBlock {
+            depends: defer_info
+                .dependencies
+                .iter()
+                .cloned()
+                .map(|id| DeferredDependency {
+                    id,
+                    defer_label: None, // TODO: is this never set?
```

Comment:
Unsure about this. In TypeScript it’s an implicit `undefined`.

Reply:
@SimonSapin 
You're right, there's an entry for it in the TS for the node, but it's never set or read. Should be fine to remove it.